python/cudf/cudf/core/buffer/utils.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+from typing import Any, Union
+
+from cudf.core.buffer.buffer import Buffer, cuda_array_interface_wrapper
+
+
+def as_buffer(
+    data: Union[int, Any],
+    *,
+    size: int = None,
+    owner: object = None,
+) -> Buffer:
+    """Factory function to wrap `data` in a Buffer object.
+
+    If `data` isn't a buffer already, a new buffer that points to the memory of
+    `data` is created. If `data` represents host memory, it is copied to a new
+    `rmm.DeviceBuffer` device allocation. Otherwise, the memory of `data` is
+    **not** copied, instead the new buffer keeps a reference to `data` in order
+    to retain its lifetime.
+
+    Parameters
+    ----------
+    data : int or buffer-like or array-like
+        An integer representing a pointer to device memory or a buffer-like
```

Comment:
> pointer to device memory

Can it also be a pointer to host memory? If so, we should clarify. If not, the docs are a bit confusing, because previously we say:

>  If `data` represents host memory, it is copied to a new `rmm.DeviceBuffer` device allocation.

Reply:
Added: 
```
If `data` is an integer, it is assumed to point to device memory.
```